### DISCUSSION TITLE
How to use docker images from Docker Desktop on WSL2 using podman?

### DISCUSSION THREAD

**Discussion by YevheniiPokhvalii** [containers/podman#16020] [General]
I am totally new to podman and I'm trying to move from docker to podman on WSL2, because podman does not depend on the deamon.  
What I discovered is that podman uses user HOME folder to store images and containers there.
By default:
```
podman info | grep graphRoot
graphRoot: /home/user/.local/share/containers/storage
```
I managed to locate Docker Desktop images in:
```
/mnt/wsl/docker-desktop-data/version-pack-data/community/docker/overlay2
```

I found somewhere in the internet that you can indicate the path to images to podman with something like that:
```
sudo podman images --root /mnt/wsl/docker-desktop-data/version-pack-data/community/docker/overlay2
```

That did not work for me. Moreover that command changed ownership of a couple of directories and I was not able to use podman without root. So I changed the ownership to the user to get it back to work as before.

It is important for me to have Docker Desktop and Podman that can work interchangeably.
Could you give any clear instructions on how to feed Docker Desktop images on WSL2 to podman? I haven't yet managed to find any working solution.

**Comment by afbjorklund** [comment]
Normally you would use something like `docker save | podman load`. You could also set up a local registry ?

**Reply by YevheniiPokhvalii** [reply]
> Normally you would use something like `docker save | podman load`. You could also set up a local registry ?

So I need to save image by image or write a script, right?
Are their configurations not compatible? Is there a solution to indicate a path to the `docker` configuration files directly to `podman`?

And what does this command actually do? It seems that it should change the folder of where the images are read from, but somehow it did not work for me and even broke some folder ownership.
```
sudo podman images --root /mnt/wsl/docker-desktop-data/version-pack-data/community/docker/overlay2
```

**Reply by afbjorklund** [reply]
There _is_ another way, with `docker-daemon:` that was indicated below.

The approach with using `save` to an archive (.tar), and then `load` from the archive, works with almost any runtime. Since it uses the CLI or API, it doesn't have to mess with the engine internals... Another compatible alternative is to use `push` and `pull`, to a registry (http server). It is more efficient, since it can negotiate which layers already exist and don't have to be transferred again. The same approach is taken when using `docker-daemon` to communicate with the DOCKER_HOST. But then dockerd needs to be running.

**Reply by afbjorklund** [reply]
> Are their configurations not compatible? Is there a solution to indicate a path to the `docker` configuration files directly to `podman`?

If by "configuration" you mean storage, then I don't think they are compatible. Only on the CLI and API level, not on the internals.

**Reply by YevheniiPokhvalii** [reply]
Thank you very much for the explanation. I'm still curious about the `sudo podman images --root` command.

**Comment by rhatdan** [comment]
Inside of the linux VM you can do podman pull docker-daemon:image

**Comment by datamike** [comment]
@YevheniiPokhvalii , I was trying to do a very similar thing while working with docker containers I had spun up in WSL Ubuntu.  I tried `docker save` , but could not get it to work.

However, this worked very nicely:

`sudo docker container export <containerid> ~/myarchive.tar`

Note that I needed to redirect the file output to my WSL home directory with '~/.'  Otherwise I got permission denied errors.

**Reply by datamike** [reply]
Now I'm going to try to import those containers into podman and see what they look like in Podman Desktop :-)

**Reply by datamike** [reply]
and for an update, `podman import` was not fruitful.   However, this post from a Red Hat Podman expert was a great find: https://www.redhat.com/sysadmin/copy-docker-images-podman

**Reply by rhatdan** [reply]
Glad it worked for you. :1st_place_medal:
